src/linelast_solver.cpp

```diff
@@ -0,0 +1,429 @@
+// Copyright 2023 Lawrence Livermore National Security, LLC. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: MIT
+
+#include "linelast_solver.hpp"
+#include "input_parser.hpp"
+#include "linalg_utils.hpp"
+#include "etc.hpp"
+
+using namespace std;
+using namespace mfem;
+
+LinElastSolver::LinElastSolver()
+    : MultiBlockSolver()
+{
+   alpha = config.GetOption<double>("discretization/interface/alpha", -1.0);
+   kappa = config.GetOption<double>("discretization/interface/kappa", (order + 1) * (order + 1));
+
+   var_names = GetVariableNames();
+   num_var = var_names.size();
+
+   // solution dimension is determined by initialization.
+   udim = dim;
+   vdim.SetSize(num_var);
+   vdim = dim;
+
+   // Set up FE collection/spaces.
+   fec.SetSize(num_var);
+   if (full_dg)
+   {
+      fec = new DG_FECollection(order, dim, BasisType::GaussLobatto);
+   }
+   else
+   {
+      fec = new H1_FECollection(order, dim);
+   }
+
+   fes.SetSize(numSub);
+   for (int m = 0; m < numSub; m++)
+   {
+      fes[m] = new FiniteElementSpace(meshes[m], fec[0], udim);
+   }
+
+}
+
+LinElastSolver::~LinElastSolver()
+{
+   delete a_itf;
+
+   DeletePointers(bs);
+   DeletePointers(as);
+   delete lambda_c;
+   delete mu_c;
+
+   delete globalMat_mono;
+   delete globalMat;
+   delete globalMat_hypre;
+   delete mumps;
+}
+
+void LinElastSolver::SetupIC(std::function<void(const Vector &, Vector &)> F)
+{
+   init_x = new VectorFunctionCoefficient(dim, F);
```

Comment:
Where do you delete `init_x`? I think this should be deleted in `~LinElastSolver`